validator/validator.ts

```diff
@@ -0,0 +1,122 @@
+// This file makes some naive assumptions surrounding the way RPC like calls will be made in ETH2.0
+/**
+ * 1. Setup any necessary connections (RPC,...)
+ * 2. Check if the chain start log has been emitted
+ * 3. Get the validator index
+ * 4. Setup block processing and attestation services
+ * 5. Wait for role change
+ * 6. Execute role
+ * 7. Wait for new role
+ * 6. Repeat step 5
+ */
+import {GenesisInfo, ValidatorCtx} from "./types";
+import RPCProvider from "./stubs";
+import {ValidatorIndex} from "../src/types";
+import BlockProcessingService from "./block";
+import {SLOTS_PER_EPOCH} from "../src/constants";
+import {getCommitteeAssignment} from "./helpers";
+import {getCurrentEpoch} from "../src/chain/helpers/stateTransitionHelpers";
+
+/**
+ * Main class for the Validator client.
+ */
+class Validator {
+  private ctx: ValidatorCtx;
+  private logger: Function;
+  private provider: RPCProvider;
+  private validatorIndex: ValidatorIndex;
+  private blockService: BlockProcessingService;
+  private genesisInfo: GenesisInfo;
+  public isActive: boolean;
+
+  /**
+   * @param {ValidatorCtx} ctx
+   */
+  public constructor(ctx: ValidatorCtx) {
+    this.ctx = ctx;
+    this.logger = (msg: string) => console.log(msg);
```

Comment:
use new logger? It's singleton so you can just import it anywhere

Reply:
Loollll I forgot about this thanks!